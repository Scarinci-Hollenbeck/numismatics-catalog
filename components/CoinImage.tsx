import React from 'react';

type Props = {
  image: string,
  title: string
};

const CoinImage = ({ image, title }: Props) : JSX. Element => <img src={image} alt={title} className="coin-image" />;

export default CoinImage;
