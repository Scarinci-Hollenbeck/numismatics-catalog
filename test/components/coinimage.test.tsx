import React from 'react';
import renderer from 'react-test-renderer';
import CoinImage from '../../components/CoinImage';

const image = [
  {
    image:
      'http://res.cloudinary.com/numismatics-catalog/image/upload/v1601061183/coins/kw2osskjam04mvgiokef.jpg',
    title: 'Coin Item 1',
  },
];

describe('CoinImage Component', () => {
  it('expect image length == 1', () => {
    expect(image.length).toBe(1);
  });

  it('The word "Coin" is found in the title', () => {
    const { title } = image[0];
    expect(title).toMatch(/Coin/);
  });

  it('the coin image list contains title and image', () => {
    expect(image[0]).toHaveProperty('title', 'Coin Item 1');
    expect(image[0]).toHaveProperty(
      'image',
      'http://res.cloudinary.com/numismatics-catalog/image/upload/v1601061183/coins/kw2osskjam04mvgiokef.jpg',
    );
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(<CoinImage image={image[0].image} title={image[0].title} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
