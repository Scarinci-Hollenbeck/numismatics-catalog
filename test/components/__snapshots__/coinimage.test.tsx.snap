// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinImage Component matches snapshot 1`] = `
<span
  className=" lazy-load-image-background blur"
  style={
    Object {
      "backgroundImage": "",
      "backgroundSize": "",
      "color": "transparent",
      "display": "inline-block",
      "height": "auto",
      "width": "100%",
    }
  }
>
  <span
    className=""
    style={
      Object {
        "display": "inline-block",
        "height": "auto",
        "width": "100%",
      }
    }
  />
</span>
`;
